src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1335,6 +1335,232 @@ class AlterTableActionSegment(BaseSegment):
+                "DEPENDS",
+                "ON",
+                "EXTENSION",
+                Ref("ParameterNameSegment"),
+            ),
+            Sequence(
+                "ALL",
+                "IN",
+                "TABLESPACE",
+                Ref("TableReferenceSegment"),
+                Sequence(
+                    "OWNED",
+                    "BY",
+                    Delimited(Ref("ObjectReferenceSegment")),
+                    optional=True,
+                ),
+                "SET",
+                "TABLESPACE",
+                Ref("ParameterNameSegment"),
+                Sequence("NOWAIT", optional=True),
+            ),
+        ),
+    )
+
+
+@postgres_dialect.segment()
+class AlterMaterializedViewActionSegment(BaseSegment):
+    """Alter Materialized View Action Segment.
+
+    Matches the definition of action in https://www.postgresql.org/docs/14/sql-altermaterializedview.html
+    """
+
+    type = "alter_materialized_view_action_segment"
+
+    match_grammar = OneOf(
+        Sequence(
+            "ALTER",
+            Ref.keyword("COLUMN", optional=True),
+            Ref("ColumnReferenceSegment"),
+            OneOf(
+                Sequence("SET", "STATISTICS", Ref("NumericLiteralSegment")),
+                Sequence(
+                    "SET",
+                    Bracketed(
+                        Delimited(
+                            Sequence(
+                                Ref("ParameterNameSegment"),
+                                Sequence(
+                                    Ref("EqualsSegment"),
+                                    Ref("LiteralGrammar"),
+                                    optional=True,
+                                ),
+                            ),
+                        )
+                    ),
+                ),
+                Sequence(
+                    "RESET",
+                    Bracketed(Delimited(Ref("ParameterNameSegment"))),
+                ),
+                Sequence(
+                    "SET", "STORAGE", OneOf("PLAIN", "EXTERNAL", "EXTENDED", "MAIN")
+                ),
+                Sequence("SET", "COMPRESSION", Ref("ParameterNameSegment")),
+            ),
+        ),
+        Sequence("CLUSTER", "ON", Ref("ParameterNameSegment")),
+        Sequence("SET", "WITHOUT", "CLUSTER"),
+        Sequence(
+            "SET",
+            Bracketed(
+                Delimited(
+                    Sequence(
+                        Ref("ParameterNameSegment"),
+                        Sequence(
+                            Ref("EqualsSegment"), Ref("LiteralGrammar"), optional=True
+                        ),
+                    )
+                )
+            ),
+        ),
+        Sequence(
+            "RESET",
+            Bracketed(Delimited(Ref("ParameterNameSegment"))),
+        ),
+        Sequence(
+            Sequence("NO", optional=True), "INHERIT", Ref("TableReferenceSegment")
+        ),
```

Comment:
where does this come from, I couldn't find it in the docs? https://www.postgresql.org/docs/14/sql-altermaterializedview.html

Reply:
Nice catch. Fixing shortly